pkg/action/scan.go

```diff
@@ -45,7 +44,7 @@ func findFilesRecursively(ctx context.Context, root string, c Config) ([]string,
 				return nil
 			}
 
-			if c.IgnoreSelf && path == self {
+			if c.IgnoreSelf && strings.Contains(path, BINARY) {
```

Comment:
I think we have multiple forms of broken behavior going on; each with their own fragility: path based and YARA based (https://github.com/chainguard-dev/bincapz/blob/main/rules/internal/bincapz.yara).

What do you think about us taking a strict approach that only ignores the file if:

- The file basename is `bincapz`
- The YARA rule matches

Reply:
The concern I have is someone reporting a "vulnerability" that they can just name their malware `bincapz` and get around the checks. At least this way they'll have to work for it.

Reply:
Very true -- I'll work on those improvements.

Reply:
Updated in [`deb9987` (#323)](https://github.com/chainguard-dev/bincapz/pull/323/commits/deb9987abba39488fbe5c48910b62d7cd8d259bf).

I tried to consolidate all of the logic in `Generate` -- the downside here is that the binary will be scanned regardless of whether we're ignoring it so it won't be as fast as skipping over it when adding it to the list of files to scan.